Assets/MixedRealityToolkit/InputSystem/Sources/BaseGenericInputSource.cs

```diff
@@ -19,12 +19,34 @@ public class BaseGenericInputSource : IMixedRealityInputSource
         private static IMixedRealityInputSystem inputSystem = null;
         public static IMixedRealityInputSystem InputSystem => inputSystem ?? (inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>());
 
+        public BaseGenericInputSource(string name, InteractionDefinition[] interactions, IMixedRealityPointer[] pointers = null)
+        {
+            SourceId = InputSystem.GenerateNewSourceId();
+            SourceName = name;
+            Pointers = pointers ?? new[] { GazeProvider.GazePointer };
+            Interactions = new Dictionary<InputType, InteractionDefinition>();
+            foreach (var interaction in interactions)
+            {
+                Interactions.Add(interaction.InputType, interaction);
+            }
+        }
+
         public BaseGenericInputSource(string name, InputType[] capabilities, IMixedRealityPointer[] pointers = null)
```

Comment:
Maybe we should get rid of the capabilities all together?

Reply:
Already done and updated the InputSources you already have after I pulled your patest changes.

Updated the Capabilities logic to use the InteractionDefinitions dictionay, which is more efficient.

But I seriously need to rethink the "Supports Capability" logic, as it returns true if ONE of the required capabilities is valid, which could cause issue.

Reply:
Fixed the support capabilities in my pr